spring-integration-core/src/main/java/org/springframework/integration/store/SimpleMessageStore.java

```diff
@@ -233,8 +270,8 @@ public void removeMessageGroup(Object groupId) {
 					return;
 				}
 
-				groupUpperBound.release(groupIdToMessageGroup.get(groupId).size());
-				groupIdToMessageGroup.remove(groupId);
+				this.groupToUpperBound.remove(groupId).release(this.groupCapacity);
```

Comment:
Agree, i checked the addMessageToGroup code to make sure. Id still be paranoid about it and add a null check to future proof the code on  this.groupToUpperBound.remove(groupId)